frontend/model/database.js

```diff
@@ -26,26 +26,43 @@ const localforage = {
               reject(new Error('Unsupported characters in name: -'))
               return
             }
-            const request = self.indexedDB.open(name + '--' + storeName)
 
-            // Create the object store if it doesn't exist
-            request.onupgradeneeded = (event) => {
-              const db = event.target.result
-              db.createObjectStore(storeName)
+            const openDB = (version?: number) => {
+              // By default `version` is the latest DB version. Initially, we
+              // try to open that, but in some cases (e.g., when manually
+              // deleting the DBs), the schema will be wrong and miss the object
+              // store. In these cases, we need to upgrade the DB by
+              // incrementing the version number to re-create the schema, which
+              // can only be done when the DB is being 'upgraded'.
+              const request = self.indexedDB.open(name + '--' + storeName, version)
+
+              // Create the object store if it doesn't exist
+              request.onupgradeneeded = (event) => {
+                const db = event.target.result
+                db.createObjectStore(storeName)
+              }
+
+              request.onsuccess = (event) => {
+                const db = event.target.result
+                if (!db.objectStoreNames.contains(storeName)) {
+                  return openDB(db.version + 1)
```

Comment:
`version` is initially `undefined`... and `undefined + 1` = `NaN`... so, is this right?

Reply:
`db.version` isn't the same as `version`.

Reply:
The reference says:

> The version property of the [IDBDatabase](https://devdocs.io/dom/idbdatabase) interface is a [64-bit integer](https://firefox-source-docs.mozilla.org/nspr/index.html) that contains the version of the connected database. When a database is first created, this attribute is an empty string.

So... 

```
> '' + 1
'1'
> '1' + 1
'11'
```

Is this what you intended?

Reply:
No, that's not what I intended, but:

1. That's not an authoritative source. <https://w3c.github.io/IndexedDB/#database-construct>, which is, mentions no empty string.
2. Even then, it wouldn't matter because when the DB is first created, we can assume that `onupgradeneeded` will run, so this check wouldn't be needed to begin with. We only need this to trigger a version upgrade.
3. Even then, going from `1` to `11` to `111` &c. should be fine.

Reply:
I've also added a logging line to `db.version` and I only ever see a number, not `''`. I'm guessing an empty string isn't used (which makes sense, since it's supposed to be an integer type).